sem/HW/main.py

```diff
@@ -51,16 +61,26 @@ def clothes():
 
 @app.post("/authentication/")
 def authentication_post():
-    name = request.form.get('username')
-    if not name:
-        flash('Введите имя!', 'danger')
-        return redirect(url_for('authentication_post'))
-    session['username'] = name
-    return redirect(url_for('html_index'))
+    form = RegistrationForm()
+    if request.method == 'POST' and form.validate():
+        full_name = f"{form.username.data.capitalize()} {form.usersurname.data.capitalize()}"
+        email = form.email.data
+        hash_password = generate_password_hash(form.password.data)
+        existing_user = User.query.filter((User.full_username == full_name) | (User.email == email)).first()
+        if existing_user:
+            error_msg = 'Username or email already exists.'
+            form.username.errors.append(error_msg)
+            return render_template('authentication.html', form=form)
+        user = User(full_username=full_name, email=email, password=hash_password)
+        db.session.add(user)
+        db.session.commit()
+        return 'Registration success!'
```

Comment:
Может лучше перенаправить сюда же
```suggestion
        return redirect(url_for('authentication_post'))
```